src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSScoreUtils.java

```diff
@@ -0,0 +1,426 @@
+            hits.add(SAMSequenceRecord.truncateSequenceName(read.getAssignedContig()));
+        }
+        hits.addAll(getValidAlternateHits(read, "XA", 0, 2, 3, minCoverage, minIdentity));
+        hits.addAll(getValidAlternateHits(read, "SA", 0, 3, 5, minCoverage, minIdentity));
+        return hits;
+    }
+
+    /**
+     * Parses alternate alignments tag of form TAG:Z:val_A0,val_A1,val_A2,...;val_B0,val_B1,val_B2,...;...
+     */
+    private static Collection<String> getValidAlternateHits(final GATKRead read, final String tag, final int contigIndex,
+                                                            final int cigarIndex, final int numMismatchesIndex, final double minCoverage,
+                                                            final double minIdentity) {
+        final Collection<String> alternateHits = new ArrayList<>();
+        if (read.hasAttribute(tag)) {
+            final int expectedTokens = Math.max(contigIndex, Math.max(cigarIndex, numMismatchesIndex)) + 1;
+            final String tagValue = read.getAttributeAsString(tag);
+            final String[] tagTokens = tagValue.split(";");
+            for (final String tok : tagTokens) {
+                final String[] subtokens = tok.split(",");
+                if (subtokens.length < expectedTokens) {
+                    throw new UserException.BadInput("Error parsing " + tag + " tag: expected at least " + expectedTokens + " values per alignment but found " + subtokens.length);
+                }
+                final String recordName = SAMSequenceRecord.truncateSequenceName(subtokens[contigIndex]);
+                final Cigar cigar = TextCigarCodec.decode(subtokens[cigarIndex]);
+                final int numMismatches = Integer.valueOf(subtokens[numMismatchesIndex]);
+                if (isValidAlignment(cigar, numMismatches, minCoverage, minIdentity)) {
+                    alternateHits.add(recordName);
+                }
+            }
+        }
+        return alternateHits;
+    }
+
+
+    /**
+     * Returns true if the candidate alignment of the read meets coverage and identity criteria
+     */
+    public static boolean isValidAlignment(final Cigar cigar, final int numMismatches, final double minCoverage, final double minIdentity) {
+        final int minCoveredBases = (int) Math.ceil(minCoverage * cigar.getReadLength());
+        int numCoveredBases = 0;
+        final List<CigarElement> cigarElements = cigar.getCigarElements();
+        for (final CigarElement cigarElement : cigarElements) {
+            if (cigarElement.getOperator().equals(CigarOperator.MATCH_OR_MISMATCH) || cigarElement.getOperator().equals(CigarOperator.INSERTION)) {
+                numCoveredBases += cigarElement.getLength();
+            }
+        }
+        final int minIdentityBases = (int) Math.ceil(minIdentity * numCoveredBases);
+        return !(new HostAlignmentReadFilter(minCoveredBases, minIdentityBases)).test(cigar, numMismatches);
```

Comment:
Not a big deal so no need to change if you don't care, but it seems like you could do this check with one pass over the CIGAR elements instead of two.